mesonpy/__init__.py

```diff
@@ -742,26 +737,15 @@ def _run(self, cmd: Sequence[str]) -> None:
 
     def _configure(self, reconfigure: bool = False) -> None:
         """Configure Meson project."""
-        sys_paths = mesonpy._introspection.SYSCONFIG_PATHS
         setup_args = [
-            f'--prefix={sys.base_prefix}',
             os.fspath(self._source_dir),
             os.fspath(self._build_dir),
             f'--native-file={os.fspath(self._meson_native_file)}',
-            # TODO: Allow configuring these arguments
-            '-Ddebug=false',
+            # default build options
+            '-Dbuildtype=release',
             '-Db_ndebug=if-release',
-            '-Doptimization=2',
-
-            # XXX: This should not be needed, but Meson is using the wrong paths
-            #      in some scenarios, like on macOS.
-            #      https://github.com/mesonbuild/meson-python/pull/87#discussion_r1047041306
-            '--python.purelibdir',
-            sys_paths['purelib'],
-            '--python.platlibdir',
-            sys_paths['platlib'],
```

Comment:
I'll note that the change to `--destdir` and these options no longer being needed was discussed in https://github.com/mesonbuild/meson-python/pull/338#discussion_r1135271706. Looks like we agreed on that, and it has a good rationale.